src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -701,6 +719,26 @@ private ResultState AssertionStatusToResultState(AssertionStatus status)
         }
 
         /// <summary>
+        /// Adds a attachments element to a node and returns it.
+        /// </summary>
+        /// <param name="targetNode">The target node.</param>
+        /// <returns>The new attachments element.</returns>
+        private TNode AddAttatchmentsElement(TNode targetNode)
+        {
+            TNode attachmentsNode = targetNode.AddElement("attachments");
+
+            foreach (var attachment in TestAttachments)
+            {
+                var attachmentNode = attachmentsNode.AddElement("attachment", attachment.Filepath);
+
+                if (attachment.Description != null)
+                    attachmentNode.AddAttribute("description", attachment.Description);
```

Comment:
Amazing how things grow! I didn't imagine we would have a description at all at the start but if we have one I think @rprouse is right.